### PR TITLE
Refactor sequencer options to make API modules modular

### LINKED ISSUE #491: Add healthcheck endpoint even without API module
When we run the sequencer API module, we get a full tide-disco server including healthcheck and version endpoints. When we opt out of the API module, we get nothing. This is problematic for cloud deployment. At minimum, the sequencer should serve a tide-disco API with the default (health and version) endpoints no matter what the config.

### PR DESCRIPTION
**Pull request by jbearer**
Previously, we had one optional module for the entire API. If you chose this you would get all the services (query service and submit API), if you omitted it you would get nothing. This makes it impossible to run an HTTP server with just a healthcheck endpoint, for example, or to run just the submit API without the heavyweight query API.

Now, there is one base module (http) which starts a barebones web server with healthcheck and version endpoints. There are additional modules you can add after that one (query and commit) which enable additional functionality.

Closes #491

### CODE REVIEW

**Comment by sveitser** [comment]
I think this may not be working as intended.

I'm getting errors when using them in the way they are now used in the docker file:

```
$  target/debug/sequencer  -- http --port 8888 -- api  
error: unrecognized subcommand 'api'

Usage: sequencer -- <COMMAND>

For more information, try '--help'.
```

After building and running the docker containers I get the same error from the sequencer nodes.

**Comment by jbearer** [comment]
Ah shoot just need to change `api` to `query` (and add `submit` to at least one of them)

**Comment by jbearer** [comment]
Should be fixed now

**Review by sveitser — APPROVED** [review]
LGTM

**Comment by sveitser** [comment]
CI failures is due to new rust release (I think). I will make a PR to fix it.

File: sequencer/src/options.rs [hunk 1]
```diff
@@ -95,8 +99,30 @@ impl ModuleArgs {
...
+
+            fn requires() -> Vec<&'static str> {
+                vec![$($($req),*)?]
+            }
+        }
+    };
+}
+
+module!("http", api::HttpOptions);
+module!("query", api::QueryOptions, requires: "http");
```
[the remark this answers is not in the document]
Comment (jbearer) [reply sequencer/src/options.rs:120]: Just comma-separated

**Pull request merged**